pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java

```diff
@@ -77,7 +79,7 @@
     private final Authentication authentication;
     private State state;
 
-    private final ConcurrentLongHashMap<CompletableFuture<Pair<String, Long>>> pendingRequests = new ConcurrentLongHashMap<>(
+    private final ConcurrentLongHashMap<CompletableFuture<Triple<String, Long, byte[]>>> pendingRequests = new ConcurrentLongHashMap<>(
```

Comment:
Maybe we could switch to a named class instead of the `Triple` so that it's easier to follow the meaning of each of them.

Reply:
Yeah, I had a similar thought but took the easy way out :-( I'll add a named class.